imageruler/imageruler.py

```diff
@@ -711,36 +711,39 @@ def get_kernel(
   """
 
   pixel_size = np.asarray(pixel_size)
-  shape = np.array(np.round(diameter / pixel_size), dtype=int)
+  kernel_size = np.array(np.round(diameter / pixel_size), dtype=int)
 
-  if shape[0] <= 2 and shape[1] <= 2:
-    return np.ones(shape, dtype=np.uint8)
+  if kernel_size[0] <= 2 and kernel_size[1] <= 2:
```

Comment:
What about non-square kernels, e.g. `kernel_size[0] == 0 and kernel_size[1] == 3`?

Reply:
This PR did not take care of this situation. The fix should be revised as something like:
```
  if kernel_size in np.array([[2,2], [2,1], [1,2], [2,1]]):
    return np.ones(kernel_size, dtype=np.uint8)

  if kernel_size[0] == 0 or kernel_size[1] == 0:
    return np.array([[1]], dtype=np.uint8)
```

Reply:
In the revised PR, I used the following lines instead:
```
  if kernel_size[0] == 0 or kernel_size[1] == 0:
    return np.array([[1]], dtype=np.uint8)
  elif kernel_size in np.array([[2,2], [2,1], [1,2], [2,1]]):
    return np.ones(kernel_size, dtype=np.uint8)
  else:
    rounded_size = np.round(diameter / pixel_size - 1) * pixel_size
```

Reply:
Do we actually support non-square kernels?

Reply:
Yes, the code including its early versions supports non-square kernels, which typically arise if a pixel has different sizes in x and y directions.